File: 4_Angular_WebAPI_MVC/EmployeeServer/EmployeeServer/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace EmployeeServer.Models
{
    public class Employee
    {
        public string EmployeeCode { get; set; }

        public string EmployeeName { get; set; }

        public double EmployeeSalary { get; set; }

        public DateTime SalesDateTime { get; set; }
    }
}

File: 5_Angular_CustomDirectives_Isolation_Restrict/Scripts/Project Scripts/Employee.js
function Employee(Utility) {
    this.EmployeeName = "Rameshkartik";
    this.EmployeeCode = "IJK1010";
    this.EmployeeSalary = 12000;
    this.SalesDateTime = Utility.getDate();
    this.CalculateDiscount = function () {
        alert("10");
    }
}


function SeniorEmployee(Utility) {
    this.EmployeeName = "Rameshkartik";
    this.EmployeeCode = "IJK1010";
    this.EmployeeSalary = 12000;
    this.SalesDateTime = Utility.getDate();
    this.CalculateDiscount = function () {
        alert("50");
    }
}

function Factory() {
    return {
        CreateEmployee: function (type,Utility) {

            if (type == 1) {
                return new Employee(Utility);
            } else {
                return new SeniorEmployee(Utility);
            }

        }
    }
    }
